packages/hardhat-core/src/builtin-tasks/help.ts

```diff
@@ -1,30 +1,75 @@
 import { HelpPrinter } from "../internal/cli/HelpPrinter";
 import { HARDHAT_EXECUTABLE_NAME, HARDHAT_NAME } from "../internal/constants";
 import { task } from "../internal/core/config/config-env";
+import { HardhatError } from "../internal/core/errors";
+import { ERRORS } from "../internal/core/errors-list";
 import { HARDHAT_PARAM_DEFINITIONS } from "../internal/core/params/hardhat-params";
 
 import { TASK_HELP } from "./task-names";
 
 task(TASK_HELP, "Prints this message")
+  .addOptionalPositionalParam(
+    "scopeOrTask",
+    "An optional task to print more info about"
+  )
   .addOptionalPositionalParam(
     "task",
     "An optional task to print more info about"
   )
   .setAction(
-    async ({ task: taskName }: { task?: string }, { tasks, version }) => {
+    async (
+      {
+        scopeOrTask: first,
+        task: second,
+      }: { scopeOrTask?: string; task?: string },
```

Comment:
The names `first` and `second` don't tell me too much about the variables and make the code a bit harder to read. I'd stick with `scopeOrTask` and `task`.